ddtrace/tracer.py

```diff
@@ -98,7 +103,10 @@ def __init__(self, url=DEFAULT_AGENT_URL, dogstatsd_url=DEFAULT_DOGSTATSD_URL):
         hostname = self.DEFAULT_HOSTNAME
         port = self.DEFAULT_PORT
         writer = None
-        if url is not None:
+
+        if _is_agentless_environment() and url == Tracer.DEFAULT_AGENT_URL:
```

Comment:
I still don't quite get why we need to check the `url` here

Reply:
I can leave this out, but just trying to check if they have explicitly set the url to something other than the default.

Reply:
Yeah I guess it comes down to how confident we are in the `_is_agentless_environment()`. I'm okay with leaving having this additional check if you feel like it's worth it 🙂